src/components/Overview.tsx

```diff
@@ -1,47 +1,121 @@
+
+  const handleSearch = (e: React.FormEvent) => {
+    e.preventDefault();
+    setIsOnView(localIsOnView);
+    setIsPublicDomain(localIsPublicDomain);
+    setIsHiddenGem(localIsHiddenGem);
+    setSearchQuery(localSearchQuery);
+    setIsDialogOpen(false);
+  };
+
+  const artworks = useFetchArtworks(
+    isOnView,
+    isPublicDomain,
+    isHiddenGem,
+    searchQuery
+  );
+
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+
+  const toggleDialog = () => {
+    if (!isDialogOpen) {
+      setLocalIsOnView(isOnView);
+      setLocalIsPublicDomain(isPublicDomain);
+      setLocalIsHiddenGem(isHiddenGem);
+      setLocalSearchQuery(searchQuery);
+    }
+    setIsDialogOpen(!isDialogOpen);
+  };
 
   useEffect(() => {
     localStorage.setItem("isOnView", JSON.stringify(isOnView));
     localStorage.setItem("isPublicDomain", JSON.stringify(isPublicDomain));
     localStorage.setItem("isHiddenGem", JSON.stringify(isHiddenGem));
-  }, [isOnView, isPublicDomain, isHiddenGem]);
+    localStorage.setItem("searchQuery", searchQuery);
+  }, [isOnView, isPublicDomain, isHiddenGem, searchQuery]);
+
   return (
     <div className="container">
-      <h1>Overview</h1>
-      <div className="filters-container">
-        <button
-          onClick={() => setIsHiddenGem(!isHiddenGem)}
-          className={isHiddenGem ? "" : "disabled-filter"}
-        >
-          Hidden gem
-        </button>
-        <button
-          onClick={() => setIsOnView(!isOnView)}
-          className={isOnView ? "" : "disabled-filter"}
-        >
-          On view
-        </button>
-        <button
-          onClick={() => setIsPublicDomain(!isPublicDomain)}
-          className={isPublicDomain ? "" : "disabled-filter"}
-        >
-          Public domain
+      <h1 className="title-and-button">
+        Overview
+        <button className="material-symbols-outlined" onClick={toggleDialog}>
+          tune
+          {isFiltersActive && <span className="active-indicator"></span>}
         </button>
-      </div>
+      </h1>
+      <Modal
+        isOpen={isDialogOpen}
+        onRequestClose={toggleDialog}
+        contentLabel="Filter Options"
+        className="modal-content"
+        overlayClassName="modal-overlay"
+      >
+        <h2 className="modal-title">Filters & Search</h2>
+        <div className="filters-container">
+          <button
+            onClick={() => setLocalIsHiddenGem(!localIsHiddenGem)}
+            className={localIsHiddenGem ? "" : "disabled-filter"}
+          >
+            Hidden gem
+          </button>
+          <button
+            onClick={() => setLocalIsOnView(!localIsOnView)}
+            className={localIsOnView ? "" : "disabled-filter"}
+          >
+            On view
+          </button>
+          <button
+            onClick={() => setLocalIsPublicDomain(!localIsPublicDomain)}
+            className={localIsPublicDomain ? "" : "disabled-filter"}
+          >
+            Public domain
+          </button>
```

Comment:
It's only three buttons, but you could make this DRYer as well:

```suggestion
          {[
            [localIsHiddenGem, setLocalIsHiddenGem, "Hidden gem"],
            [localIsOnView, setLocalIsOnView, "On view"],
            [localIsPublicDomain, setLocalIsPublicDomain, "Public domain"]
          ].map(([value, setter, text]) => (
            <button
              key={text}
              onClick={() => setter(!value)}
              className={value ? "" : "disabled-filter"}
            >
              {text}
            </button>
          ))}
```